### PR TITLE
[GUI Editor] Overhaul selection and simplify workbench

### PR DESCRIPTION
**Pull request by darraghjburke**
Completely reworks the selection logic to follow a simple new system:
1. On every pointer down, collect all controls that were clicked
2. On the next frame, determine which control, if any, to select. If we didn't hit any controls, we should deselect everything. If we double clicked, we should go down one level in the hierarchy. If we clicked the control or a sibling of the control, select that one. Otherwise, if we hit a root level control, select that.

There is one special case: if you have a control already selected, then selection processing part will only happen after you release the pointer. This is because you may want to drag the selected control(s) rather than selecting, so if you move your mouse more than a few pixels, you will get a drag behavior instead.

Also reduces some of the complexity of workbench.tsx and removes some bad practices. More work is definitely still need here, but it's a step in the right direction.

### CODE REVIEW

**Comment by darraghjburke** [comment]
> Want to merge this into the new structure?

Did I miss something? I think this PR should be ready to merge into master

**Review by RaananW — COMMENTED** [review]
Just a small question and LGTM!

File: packages/tools/guiEditor/src/diagram/workbench.tsx [hunk 1]
```diff
@@ -738,29 +649,94 @@ export class WorkbenchComponent extends React.Component<IWorkbenchComponentProps
...
+        this._rootContainer.current?.setPointerCapture(evt.pointerId);
 
-        const pos = this.getScaledPointerPosition();
         if (this._forceSelecting) {
-            this._mouseStartPointX = pos ? pos.x : this._mouseStartPointX;
-            this._mouseStartPointY = pos ? pos.y : this._mouseStartPointY;
+            this._mouseStartPoint = this.getScaledPointerPosition();
+        }
+
+        if (evt.button !== 0 || this._forcePanning) {
```
[the remark this answers is not in the document]
Comment (darraghjburke) [reply packages/tools/guiEditor/src/diagram/workbench.tsx:708]: Yes, the idea is that middle mouse or right mouse should pan the canvas. Let me double check with Patrick

[the remark this answers is not in the document]
Comment (RaananW) [reply packages/tools/guiEditor/src/diagram/workbench.tsx:708]: discussed, will be addressed in a future PR

**Review by RaananW — APPROVED** [review]

**Pull request merged**